File: nvim/lua/core/lastline.lua
vim.opt.laststatus = 0
vim.api.nvim_set_hl(0, 'StatusLine', { link = 'Normal' })
vim.api.nvim_set_hl(0, 'StatusLineNC', { link = 'Normal' })
local width = vim.api.nvim_win_get_width(0)
local line = string.rep('─', width)
vim.opt.statusline = line

File: nvim/oxi/core/src/lastline.rs
use std::usize;

use nvim_oxi::{
    api::{self, opts::SetHighlightOpts},
    Error,
};

pub fn init() -> Result<(), Error> {
    api::set_option("laststatus", 0)?;
    let opt = SetHighlightOpts::builder().link("Normal").build();
    api::set_hl(0, "StatusLine", &opt)?;
    api::set_hl(0, "StatusLineNC", &opt)?;
    let width = api::get_current_win().get_width()? as usize;
    let line = "─".repeat(width);
    api::set_option("statusline", line)?;
    Ok(())
}
